Aloha update

TomWoodward opened the pull request:
encountered some problems with the note plugin that @cprerovsky thinks we can solve by updating the aloha version - probably not a terrible idea in any case

draftkraft committed:
Added the subsub event "aloha.link.insert" after a link is inserted. Added a highlightClass property to add a class when a link href is edited

draftkraft committed:
Added comfort method to get the Aloha Editor content via jQuery

draftkraft committed:
Fixed scope activation

laktek committed:
Disabled table resizing for IE7. Updated the guide too

laktek committed:
Fixed the mssing variable reference

Commit:
updated guides and added button for table resizing

Commit:
updated guides and added button for table resizing

Commit:
updated table plugin button layout, internationalization and guides

Commit:
fixed rebase onto hotfix errors, added changelog and reverted boilerplate
demo

Commit:
fixed rebase onto hotfix errors

Commit:
Merge branch 'hotfix-table-resize' of github.com:cprerovsky/Aloha-Editor into hotfix-table-resize

Commit:
implemented tobis suggestions for a cleaner pull request

Commit:
Updated dev branch artifact version.

Commit:
Updated artifact version for next release

deliminator committed:
Fixes placeholders

deliminator committed:
Adds warning about using reserved classes

Commit:
Refactor RepositoryManager

Fixes numerous issues, including:
 * how query() and getChildren() handle immediate, and asynchronous
   repositories.
 * incorrect documentation
 * bad code

Commit:
Refactor RepositoryManager (minor--no functional changes)

Commit:
Update repository browser to fix bug in folder tree

The repository browser had a bug in how it listed repository folders on the tree
when working with multiple repositories--it placed each document multiple times
into every repository rather than only into the repository to which it belong.

Update to repository browser includes refactoring done to it as well.

See: repository-browser a4976e220aa3108598cc3da760602f16d66e3830

Commit:
Update repository browser

Commit:
Clear timeout in repository manager before reporting

deliminator committed:
Fixes Aloha.settings.locale property is not honored

Commit:
Provide outerHTML polyfill

Basic support for the element.outerHTML property does not exist in browsers
older than:

Firefox (Gecko) 11
Chrome 0.2
Internet Explorer 4.0
Opera 7
Safari 1.3

There are other places in Aloha Editor code where element.outerHTML support is
being assumed--where the polyfill should be used instead.  This remains an open
issue.

Commit:
Refactor comments in core (minor, non-functional changes)

Commit:
Coalesce <a> fragments during link insertion

Successive anchor elements are generally not merged in Aloha Editor for obvious
reasons, but an exception needs to be made when inserting new links: fragments
of a new link are coalesced whenever possible.

Commit:
Don't instantiate XMLSerializer in unsupportive browsers

Commit:
Dereference outerHTML once

Commit:
Merge branch 'hotfix' of github.com:alohaeditor/Aloha-Editor into release

Commit:
Merge branch 'hotfix-table-resize' of github.com:cprerovsky/Aloha-Editor into hotfix-table-resize

Commit:
Merge branch 'hotfix-table-resize' of git://github.com/cprerovsky/Aloha-Editor into hotfix

Commit:
Fixed word-wrap issue within ie8 and chrome.

Commit:
Merge branch 'hotfix' of github.com:alohaeditor/Aloha-Editor into release

Commit:
Fix bugs and enhance usability in image-plugin

Changes:
* Keep Aspect Ratio now works from the start if activated
* Keyboard-inputs for height/width are no longer auto-corrected when
  below minimum to avoid weird changes while still typing
* Toggling aspect ratio in crop-mode no longer breaks the plugin
* Field values are correctly updated after cropping and cancelling crop
* Gradually increasing size now works while keep aspect ratio is activated
* Several documentation fixes and minor refactoring

Commit:
Fix image-demo by adding necessary paths

Commit:
Fix bugs when resizing while keeping aspect-ratio

* Fixed problems when resizing and releasing mouse inside image
* Repaired toggle-button functionality

Commit:
Squashed commit of the following:

commit 08e0b6156793f0d3998ce8b024e63fcdfd60ce62
Author: Clemens Prerovsky <c.prerovsky@gentics.com>
Date:   Mon Jan 7 16:27:59 2013 +0100

    made internationalization work

commit 94bd9dce75d8c2c49211e0a33bf4090d5b90fce8
Author: Clemens Prerovsky <c.prerovsky@gentics.com>
Date:   Mon Jan 7 16:27:52 2013 +0100

    made internationalization work

commit 1f2f05f15ebec612645437ccb53f893744468f34
Merge: 91a00c9 185e0fb
Author: Clemens Prerovsky <c.prerovsky@gentics.com>
Date:   Mon Jan 7 16:13:05 2013 +0100

    Merge branch 'hotfix-textcolor-plugin' of github.com:cprerovsky/Aloha-Editor into hotfix-textcolor-plugin

commit 91a00c927d41a3c52b5a5a0119f4ed2b3abd2b64
Author: Lakshan Perera <lakshan@web2media.net>
Date:   Wed Jan 2 22:41:46 2013 +0530

    Added the colorpicker plugin

commit 185e0fb371fad77b39aed933cee02dd12787185b
Author: Lakshan Perera <lakshan@web2media.net>
Date:   Wed Jan 2 22:41:46 2013 +0530

    Added the colorpicker plugin

deliminator committed:
Fixes lint error

deliminator committed:
Merge branch 'dev' of ssh://github.com/alohaeditor/Aloha-Editor into dev

Commit:
Merge branch 'hotfix' of git://github.com/maschonber/Aloha-Editor into release

Commit:
Prepared changelog for release

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.22.7

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.22.7-commercial

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Fix internationalization in confirm-dialog

Commit:
Fix unsensible condition

Commit:
Enhance and fix characterpicker-plugin

* Overlay shows characters in the font-family/weight/style of the selected
  text
* If multiple characters are inserted, plugin now inserts them at the
  correct position
* The same focus is now used for mouse and keyboard commands
* Redesign of the CSS (making it smaller and removing white space)
* Adds a note in the guide concerning incorrectly displayed characters

Commit:
Recompiling aloha.css to include characterpicker updates

Commit:
Merge branch 'release' of github.com:alohaeditor/Aloha-Editor into hotfix

Conflicts:
	build/hotfix-changelog.md

Commit:
Refactor entire characterpicker plugin

Commit:
Add parameter annotation

Commit:
Improve inline documentation

Commit:
Fix placement of selection after inserting special character

Commit:
Lint dialog.js

Commit:
Correct spelling typos

Commit:
Add modal dialog

evo42 committed:
fixed typo in contenthandler guide

deliminator committed:
Merge pull request #857 from evo42/hotfix

fixed typo in contenthandler guide

Commit:
Have Dialog.modal() return DOM Element for flexibility

Also correct changelog entry.

evo42 committed:
fixes #841 -- "Image Insert Breaks on FireFox" (there was a problem at getPluginUrl in core.js)

deliminator committed:
Refactors the format plugin

Behaviour was not changed. Only restructuring and removal of duplicate code.

deliminator committed:
Improves b/strong, i/em handling

evo42 committed:
fixes the query method in linklist.js to show the correct links for each folder

evo42 committed:
Merge pull request #863 from evo42/hotfix

fixes #841 -- "Image Insert Breaks on FireFox"
fixes incorrect query method in linklist.js

evo42 committed:
updated a wrong file name in the README for the gengo.com import/export

evo42 committed:
Merge pull request #864 from evo42/hotfix

updated a wrong file name in the README for the gengo.com import/export

deliminator committed:
Fixes javascript error

deliminator committed:
Implements state override for the format plugin

Commit:
Alohafy tables in editables that are in blocks

deliminator committed:
Implements range-context.js

Commit:
Adds Changelog for 25d297bd8665f7f9b4752d4dbe48956fe31a39f7

Commit:
Merge branch master into dev

Conflicts:
	build/dev-changelog.md
	pom.xml
	src/css/aloha.css
	src/img/base.png
	src/lib/aloha/core.js
	src/plugins/common/format/lib/format-plugin.js
	src/plugins/common/ui/img/format-inline.png

Commit:
Linting

deliminator committed:
Merge branch 'dev-state-override' into dev

deliminator committed:
Merge branch 'dev-range-context' into dev

deliminator committed:
stableRange() should always create a new range

Because it is a mutable datastructure.

deliminator committed:
range-context - refactors range preservation

Offset based range preservation is complicated and
error prone. Instead use node/atEnd representation.

Also moved preservation code from dom2.js to
range-context.js because it is too specialized.

deliminator committed:
Merge branch 'dev-range-context' into dev

deliminator committed:
Fixes contenthandler -> table plugin dependency

We want to be able to use the contenthandler plugin
without also needing to load the table plugin.

Commit:
Merge branch 'dev' of github.com:alohaeditor/Aloha-Editor into commercial-hotfix-batu-fr1

Conflicts:
	src/css/aloha.css

Commit:
Updated aloha.css after merge conflict by building it.

Commit:
Updated artifact version

Commit:
Merge branch 'commercial-release-batu-fr1' of github.com:alohaeditor/Aloha-Editor into commercial-hotfix-batu-fr1

Commit:
Fix setting scope when formatting rows as header

Jotschi committed:
Merge pull request #875 from maschonber/commercial-hotfix-batu-fr1

Fix setting scope when formatting rows as header

Commit:
User util/browser to check for ie7

Commit:
Correctly determine when to float below an editable

Commit:
Merge branch 'commercial-hotfix-batu-fr1' of github.com:alohaeditor/Aloha-Editor into commercial-release-batu-fr1

Commit:
Update inline documentation

Commit:
Use new util/keys to determine keyCode tokens

Commit:
Updated css and prepared changelog for release

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.0-commercial

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Prepared for gpl release

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.0

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Merge branch 'commercial-hotfix-batu-fr1' of github.com:alohaeditor/Aloha-Editor into commercial-hotfix-batu-fr1

Commit:
Corrected aloha editor version.

Commit:
Publish messages on important link events

Allows monitoring of changes to links via the PuSub system.

deliminator committed:
Implements state-override without engine.js

The engine.js implementation was more difficult to extend
than to simply implement it myself from scratch.

A nice sideeffect is that it's now much faster.

The state-override feature is now turned on by default.

Conflicts:
	build/dev-changelog.md

deliminator committed:
Implements StateOverride.set()

Commit:
Merge branch 'commercial-release-batu-fr1' of github.com:alohaeditor/Aloha-Editor

Conflicts:
	build/hotfix-changelog.md
	pom.xml

Commit:
Expose internal state defensively to make it read-only

Commit:
Re-fix link creation broken in automatic merge

Link creation fix bbba6a233336d5f8a6ed217ff233da3acd262c04 was
broken during automatic merge 091bf394ec598c5bca9e417c58ff4ee76b9fc288

Commit:
Remove accidental `debugger' directive

Commit:
Lint keys.js

deliminator committed:
Implements RangeContext.splitBoundary

deliminator committed:
Merge branch 'hotfix-range-context' into hotfix

deliminator committed:
Fixes RangeContext.splitBoundary test on IE7

Commit:
Merge branch 'commercial-release-batu-fr1' of github.com:alohaeditor/Aloha-Editor into commercial-hotfix-batu-fr1

Conflicts:
	build/hotfix-changelog.md
	pom.xml

Commit:
Link plugin: Removed CSS code (moved to aloha-core)
RT#53768

Commit:
Cite plugin: Changed divs to labels and removed CSS code (moved to aloha-core)
RT#53768

Commit:
Sidebar: Enhanced the styling in all browsers allot
RT#53768

Commit:
Fixed jslint error

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.1-commercial

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Reverted release.

Commit:
Prepared changelog

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.1-commercial

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.1

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

deliminator committed:
Implements RangeContext.formatStyle

deliminator committed:
Refactors RangeContext.formatStyle

deliminator committed:
Implements ancestor reuse for RangeContext.formatStyle

deliminator committed:
Fixes RangeContext.formatStyle tests in IE7

deliminator committed:
Adds utility functions

deliminator committed:
Fixes jslint errors

deliminator committed:
Merge branch 'hotfix-range-context' into hotfix

deliminator committed:
Fixes recursive dependency

deliminator committed:
Fixes IE7 crash

Commit:
Remove cursor-styles in table after editing

Commit:
Rewrite metaview plugin

	Does the following:
	♣  Lints plugin
	♥  Makes code more readable
	♠  Protects "private" fields
	♦  Correctly reenables meta-view appropriately on re-activated editable

Commit:
Add documentation to metaview functions

Commit:
Rewrites wai-lang plugin

Commit:
Publish 'aloha.editable.activated' on PubSub system

deliminator committed:
Fixes IE7 javascript error

deliminator committed:
Fixes clicks on Aloha UI makes the toolbar disappear

deliminator committed:
Fixes higlight background color is somtimes not removed

Commit:
Improve documenations for wai-lang plugin

Commit:
Don't auto append <br>s to block-level elements unless empty

Commit:
Changelog 0.23.1: Added missing '-'

Commit:
Transfered 2 changelog entries from hotfix-changelog and enhanced them

Commit:
Enhanced changelog

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.2-commercial

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Fixed handling of closing angle brackets in the DOM to XHTML plugin

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.2

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Merge branch 'commercial-hotfix-batu-fr1' of git://github.com/HybridSystem/Aloha-Editor into commercial-hotfix-batu-fr1

Conflicts:
	build/hotfix-changelog.md

Commit:
Character plugin: Fixed font color

Commit:
Character plugin: Fixed "has no method 'is'"-error RT#54057

Commit:
aloha.css: Re-generated

Commit:
Refactoring; remove unnecessary guards

Commit:
Remove unneeded wia-lang code

Commit:
Remove erroneously commited file

Commit:
Update guide concerning changes to handling of trailing <br>s in IE

Commit:
Add warning in guides concerning multiple trailing <br>s in IE

Commit:
Clean up line break handling on editable initialization in IE

Commit:
Ensure tab is shown when in wai-lang context

Also use named functions for better easier debugging.

Commit:
Add missing changelog for 2e4d2152ceccd113b01e3460e4101e8df8c348cf

Commit:
Refactor content handler manager

Commit:
Don't compute wrapContent() if unnecessary

Commit:
Stops deleting last <br> in editable content

Also completes list of block-level elements

Commit:
Add changelog for 7fc8adc037283fba055a7af7959e2159701e5a3b

Commit:
Release: Preparing changelog

Commit:
Refactor removeTrailingBr() blockelementcontenthandler.js

Move generic operation to html.js

Commit:
Merge branch 'hotfix' into commercial-hotfix-batu-fr1

Conflicts:
	build/hotfix-changelog.md
	pom.xml
	src/css/aloha.css
	src/lib/util/html.js

Commit:
Merge branch 'commercial-hotfix-batu-fr1' of github.com:alohaeditor/Aloha-Editor into commercial-hotfix-batu-fr1

Conflicts:
	src/lib/util/html.js

Commit:
Sidebar: Added missing changelog entry for fix

Commit:
Fixed some style, escaping & indentation problems

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.3-commercial

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.3

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Added documentation for the Sidebar

Commit:
Refactor minor changes in documentation

Commit:
Fix wrong function signature of copyStyle()

Commit:
Correct use and implementation of calculateOffset()

Commit:
Correct identifier typo

Incorrectly referencing `event` instead of `$event` while determining the target
of a click event on the document body resulted in the character picker overlay
always being hidden after every mouse click in IE--even if the event target was
the characterpicker "open" button.

Commit:
Remove useless code

Commit:
Revert "Fix wrong function signature of copyStyle()"

Defer this improvement for a non-hotfix release.

This reverts commit 6ec76b0cd7819044ae7f6f09f8dc29b4f2bb32f6.

Commit:
Revert "Correct use and implementation of calculateOffset()"

Defer this improvement for a non-hotfix release.

This reverts commit 7faabc27dcd0700ae240a21f1e8d884f9a7c43a4.

Commit:
Revert "Remove useless code"

Defer this improvement for a non-hotfix release.

This reverts commit 8e13eacb46f5c868b7c34efe286d370ea70974fb.

Commit:
Reinsert "Fix wrong function signature of copyStyle()"

This reverts commit 93b49f500953d1525cb8c2edbb95cc2709e32bc4.

Commit:
Reinsert "Correct use and implementation of calculateOffset()"

This reverts commit 635bb917279223a4bc05bf967259cfc896fe4b7a.

Commit:
Reinsert "Remove useless code"

This reverts commit e129f644b78a0bee4ccd7a2c92464138f40bb189.

Commit:
Clean up depedencies

Commit:
Refactor headerids plugin methods for readability and correctness

Refactoring done, and documentation added to the methods check(), process(), and
sanitize()

Commit:
Fix table styles and activation of style-buttons

* Formatting columns, rows and cells now work in combination
* Activation-status of style-buttons now correctly displayed

Commit:
Add several table styles to boilerplate-css

Commit:
Don't change ids of headings that are Editables or Blocks

Heading elements which are Aloha editables or blocks must not have their ids
changed because implementations often use the ids of these elements to track
them in the DOM.

Commit:
naturalfit button resets width on th elements

Commit:
Merge pull request #895 from sebastianvogel/commercial-hotfix-batu-fr1-sv-tableresizefix

naturalfit button resets width on th elements

Commit:
Prepared commercial release of aloha editor.

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.4-commercial

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.4

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Minor code-cleanup for table-styles-fix

Commit:
Correct HeaderIds plugin guide

Commit:
Added missing german translations for table plugin

Commit:
Stop appending elements to rows after rowspans

Commit:
Fix cell-selection after changing header status

Commit:
Fix header-selection bugfix

Commit:
Allow the possibility to complete empty an editing hosts

Updated deleting operations to allow the possibility to completely empty an
editing host of its contents (ie: CTRL+A, DEL).

Commit:
Lint recent code changes in engine.js

Commit:
Enable deletion of solo <p><br></p> in editable

Calling delete on the given markup:
<editable><block><br>[]</block></editable>
should results in:
<editable>[]</editable>

Commit:
Allow split of single active cell with row/colspan

Commit:
Declare `pos` only once

Commit:
Add TextColor plugin guides

Taken from c988a742f92eba81f962161cb7bf9b3fbfc5a730

Commit:
Merge branch 'commercial-hotfix-batu-fr1' of github.com:alohaeditor/Aloha-Editor into hotfix-table-spanappend

Conflicts:
	build/hotfix-changelog.md

Commit:
Merge branch 'hotfix-table-header-selection' of git://github.com/maschonber/Aloha-Editor into commercial-hotfix-batu-fr1

Conflicts:
	build/hotfix-changelog.md

Commit:
Merge branch 'hotfix-table-splitactivecell' of git://github.com/maschonber/Aloha-Editor into commercial-hotfix-batu-fr1

Conflicts:
	build/hotfix-changelog.md

Commit:
Merge branch 'hotfix-table-styles' of git://github.com/maschonber/Aloha-Editor into commercial-hotfix-batu-fr1

Conflicts:
	build/hotfix-changelog.md
	src/plugins/common/table/lib/table-plugin.js

Commit:
Fixed merge error

Commit:
Prepared commercial release

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.5-commercial

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.5

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Fix inserting linebreaks in editable spans

Commit:
Don't accidentally delete activated block w/backspace
when event originates from UI

Commit:
Merge pull request #906 from maschonber/hotfix-span-break

Fix inserting linebreaks in editable spans

Commit:
Added changelog

Commit:
Prepared commercial release

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.6-commercial

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

GenticsDev committed:
[maven-release-plugin] prepare release alohaeditor-0.23.6

GenticsDev committed:
[maven-release-plugin] prepare for next development iteration

Commit:
Don't stall on disabled plugins during initialization

Commit:
Proceed with initialization when no plugins are configured

TomWoodward committed:
Merge remote-tracking branch 'upstream/commercial-hotfix-batu-fr1' into aloha-update

Conflicts:
	.gitignore
	build/dev-changelog.md
	src/lib/aloha/block-jump.js
	src/lib/aloha/command.js
	src/lib/aloha/console.js
	src/lib/aloha/contenthandlermanager.js
	src/lib/aloha/core.js
	src/lib/aloha/editable.js
	src/lib/aloha/engine.js
	src/lib/aloha/ephemera.js
	src/lib/aloha/jquery.aloha.js
	src/lib/aloha/markup.js
	src/lib/aloha/observable.js
	src/lib/aloha/plugin.js
	src/lib/aloha/pluginmanager.js
	src/lib/aloha/repositorymanager.js
	src/lib/aloha/selection.js
	src/lib/aloha/sidebar.js
	src/plugins/common/image/lib/image-plugin.js
	src/plugins/common/image/nls/i18n.js
	src/plugins/common/ui/lib/tab.js
	src/plugins/common/undo/lib/undo-plugin.js

TomWoodward committed:
fix contenthandler js error

TomWoodward commented:
dont merge this quite yet, still working on math rendering with the update

philschatz commented:
Don't worry, there's **ZERO**  chance of me getting through all these commits today ; )

TomWoodward committed:
fix syntax error in sandbox/index.html

TomWoodward committed:
add mathml elment whitelist to the sanitizer on plugin load

TomWoodward committed:
totally remove sanitization contenthandler

izak committed:
Fixed popover.hide not to break chaining.

This means that the monkey-patched destroy() was never necessary,
so this has been dropped too.

izak committed:
don't let mousedown on actions propagate

This prevents a click on the toolbar from deactivating the current
editor.

izak committed:
mousedown on popover does not propagate

This prevents the event from reaching aloha and causing the editor
to deactivate, which allows clicking on popovers without the popover
closing on you.

izak commented:
I added a couple of fixes myself, needed to make popovers and the toolbar work with the updated code.

izak committed:
Hide popover before destroying it.

Otherwise it thinks it is visisble and refuses to show up again on
click.

izak committed:
Namespace the show and hide events so they clean up.

The show and hide events, bound via delegation to the editor itself, did not
clean up properly, so there was this ever growing list of hide/show events on
every math element, which likely caused severe performance issues. Adding
a namespace ensures that it is cleaned up when .bubble is off()'ed.

izak committed:
Formatting change in comments, no functionality.  dead code cleanup

izak committed:
Replace aloha-editable-activated with -created.

Also unify the various event handlers all bound to the same event into one.
Use -created event rather than -activated, this code need only run once,
and in our old code where the editor was never deactivated, that is how it
worked anyway.

izak committed:
Avoid binding multiple mousedown handlers to body.

izak committed:
Avoid rebinding deprecated focus/blur callbacks more than once.

izak committed:
Delay cleanup of delegated event handlers on deactivation till last

If you trigger 'hide' after you've cleaned up the event handler, it's not
going to work is it?

izak committed:
Don't glue link popover to math tooltips

Specifically, the X that removes math is an <a>, but it isn't a link. This is
a generic fix that won't place popovers on anything that is marked as ephemera.

bnwest committed:
refs #374: have ephemera remove the data-original-title attribute that bootstraps inserts for tooltips.

bnwest committed:
add comment
